client/src/components/LogInModal.jsx

```diff
@@ -42,49 +38,77 @@ const LogInModal = ({ isOpen, onClose }) => {
                 const userStatus = await response.json();
 
                 //If there is a user with this username, log them in
-                if (userStatus.exists) {
-                    
+                if (userStatus.exists && userStatus.authorized) {
                     //Set the logged user state to the user id we will need to get the correct data
                     setLoggedUser({userId: userStatus.userId, name: userStatus.name});
+                    
+                    //Store access token locally to be used for authorization of all server calls
+                    localStorage.setItem("token", userStatus.accessToken);
 
                     //Switch to dashboard display
                     navigate('/dashboard');
                     
-                    //Potentially use to combat losing user on refresh:
-                    // localStorage.setItem("user", userStatus.userId);
-
                     //Close the modal
                     onClose();
                 } else {
-                    //Client side error handling to ask them to try again - need to update once log in is fleshed out
-                    console.log("Username does not exist.");
+                    setFormErrorMessage("Invalid username or password.");
                 }
             } catch (error) {
-                //Update this error handling as well once log in is complete
-                console.error({ message: "Error checking user name", details: error });
+                setFormErrorMessage("Please try again.");
```

Comment:
If you are sending back user friendly error messages, you can always just use those instead of recreating another error message. Something to keep in mind.

Reply:
Hmm I hadn't thought of that. I marked it down as something to chat about when we meet!